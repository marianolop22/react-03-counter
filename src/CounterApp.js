import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ( { value=10 } ) => {

    const [ counter, setCounter ] = useState(value);


    //handleAdd
    const handleAdd = (e) => {
        // 2 maneras de manejar el conrador
        setCounter(counter + 1);
        //setCounter( (c) => c + 1 );
    };

    const handleReset = (e) => {
        // 2 maneras de manejar el conrador
        setCounter(value);
        //setCounter( (c) => c + 1 );
    };

    const handleSubstract = (e) => {
        // 2 maneras de manejar el conrador
        setCounter(counter - 1);
        //setCounter( (c) => c + 1 );
    };


    const handle = ( value ) => {
        // 2 maneras de manejar el conrador
        console.log('value', value);
        setCounter(counter - 1);
        //setCounter( (c) => c + 1 );
    };

    


    
    return (
        <>
            <h1>Counter App</h1>
            <h2>{counter}</h2>

            {/* una opcion para mandar el parametro  <button onClick={ (e) => { handle(value) } }>+1</button> */}
            <button onClick={ handleAdd }>+1</button>
            <button onClick={ handleReset } >Reset</button>
            <button onClick={ handleSubstract } >-1</button>
        </>
    );
};


CounterApp.propTypes = {
    value: PropTypes.number
};

export default CounterApp;


