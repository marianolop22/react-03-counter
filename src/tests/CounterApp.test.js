import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';

import CounterApp from '../CounterApp';

describe ('pruebas sobre CounterApp', ()=> {

    let wrapper = shallow(<CounterApp />);
    
    beforeEach ( () => {
        wrapper = shallow(<CounterApp />);
    })

    test('debe mostrar bien el <CounterApp />', () => {
        expect( wrapper ).toMatchSnapshot()
    });

    test('debe mostrar bien el valor por defecto de 10', () => {
        
        const value = 50;
        const wrapper = shallow(<CounterApp value={value}/>);
        
        const textoContador = wrapper.find('h2').text().trim();

        expect( parseInt(textoContador) ).toBe(value);
    });

    test('debe incrementar con el boton de +1', () => {
        
        wrapper.find('button').at(0).simulate('click'); //esto me da el primer botton
        const textoContador = wrapper.find('h2').text().trim();
        expect( parseInt(textoContador) ).toBe(11);
    });

    test('debe incrementar con el boton de -1', () => {
        
        wrapper.find('button').at(2).simulate('click'); //esto me da el primer botton
        const textoContador = wrapper.find('h2').text().trim();
        expect( parseInt(textoContador) ).toBe(9);
    });

    test('debe resetear al valor por default', () => {
        
        const value = 50;
        const wrapper = shallow(<CounterApp value={value}/>);

        wrapper.find('button').at(0).simulate('click'); //esto me da el primer botton
        wrapper.find('button').at(0).simulate('click'); //esto me da el primer botton
        wrapper.find('button').at(0).simulate('click'); //esto me da el primer botton

        wrapper.find('button').at(1).simulate('click'); //esto me da el primer botton
        const textoContador = wrapper.find('h2').text().trim();
        expect( parseInt(textoContador) ).toBe(value);
    });

});