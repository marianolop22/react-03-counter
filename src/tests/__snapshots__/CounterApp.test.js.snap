// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre CounterApp debe mostrar bien el <CounterApp /> 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
